src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -0,0 +1,158 @@
+package org.broadinstitute.dsde.workbench.sam.openam
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.sam.errorReportSource
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.db.DbReference
+import org.broadinstitute.dsde.workbench.sam.db.SamParameterBinderFactory._
+import org.broadinstitute.dsde.workbench.sam.db.tables._
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.util.DatabaseSupport
+import org.postgresql.util.PSQLException
+import scalikejdbc._
+
+import scala.concurrent.ExecutionContext
+
+class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
+                              protected val ecForDatabaseIO: ExecutionContext)(implicit executionContext: ExecutionContext) extends AccessPolicyDAO with DatabaseSupport with LazyLogging {
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+
+  // Create resourceType should do the following:
+  // 1. Create the resource type
+  // 2. Add the roles
+  // 3. Add the actions
+  // 4. Add patterns
+  def createResourceType(resourceType: ResourceType): IO[ResourceType] = {
```

Comment:
A problem to address is that this function needs to do the right think if the resource type already exists. The "right thing" is definitely not to raise an exception (as the current code will do). The "right thing" probably includes making the action patterns be correct, adding any new roles, adding any new actions, updating role-actions. Perhaps remove roles that no longer exist if they are not in use. Essentially everything that needs to happen to make the database state look like what is in config. And keep in mind that if more than 1 sam starts at the same time there will be more than one client trying to do this to be resilient to concurrency issues.

If it makes sense to merge this PR and then work on the above, that is fine.